runtime/src/main/java/org/corfudb/runtime/exceptions/TransactionAbortedException.java

```diff
@@ -28,22 +34,28 @@
      */
     public TransactionAbortedException(
             TxResolutionInfo txResolutionInfo,
-            Integer conflictKey, AbortCause abortCause) {
-        this(txResolutionInfo, conflictKey, abortCause, null);
+            Integer conflictKey, AbortCause abortCause, AbstractTransactionalContext context) {
+        this(txResolutionInfo, conflictKey, null, abortCause, null, context);
     }
 
     public TransactionAbortedException(
             TxResolutionInfo txResolutionInfo,
-            Integer conflictKey, AbortCause abortCause, Throwable cause) {
+            Integer conflictKey, UUID conflictStream,
+            AbortCause abortCause, Throwable cause, AbstractTransactionalContext context) {
```

Comment:
retain the context as part of the exception and also change the toString of the AbstractTransactionalContext to print out more information.